experimental/devices/cap1188/cap1188.go

```diff
@@ -0,0 +1,554 @@
+	// state of the LED pins that are not linked to sensor inputs.
+	reg_LEDOutputControl = 0x74
+)
+
+// Dev is a handle to a cap1188.
+type Dev struct {
+	d          conn.Conn
+	regWrapper mmr.Dev8
+	isSPI      bool
+	opts       *Opts
+	mu         sync.Mutex
+}
+
+func (d *Dev) String() string {
+	return fmt.Sprintf("cap1188{%s}", d.d)
+}
+
+// Halt is a noop for the cap1188.
+func (d *Dev) Halt() error {
+	return nil
+}
+
+// InputStatus reads and returns the status of the 8 inputs as an array where
+// each entry indicates a touch event or not.
+func (d *Dev) InputStatus() ([]TouchStatus, error) {
+	// read inputs
+	status, err := d.regWrapper.ReadUint8(0x3)
+	if err != nil {
+		return inputStatuses, wrap(fmt.Errorf("failed to read the input values - %s", err))
+	}
+
+	// read deltas (in two's complement, capped at -128 to 127)
+	deltasB := [nbrOfLEDs]byte{}
+	if err = d.regWrapper.ReadStruct(0x10, &deltasB); err != nil {
+		return inputStatuses, wrap(fmt.Errorf("failed to read the delta values - %s", err))
+	}
+	deltas := [nbrOfLEDs]int{}
+	for i, b := range deltasB {
+		deltas[i] = twoComplementsToInt(b)
+	}
+
+	// read threshold
+	thresholds := [nbrOfLEDs]byte{}
+	if err = d.regWrapper.ReadStruct(0x30, &thresholds); err != nil {
+		return inputStatuses, wrap(fmt.Errorf("failed to read the threshold values - %s", err))
+	}
+
+	// convert the data into a sensor state
+	var touched bool
+	for i := uint8(0); i < nbrOfLEDs; i++ {
+		touched = isBitSet(status, 7-i)
+
+		// TODO: check if the event is passed the threshold:
+		// deltas[i] > int(thresholds[i])
+
+		if touched {
+			if inputStatuses[i] == PressedStatus {
+				if d.opts.RetriggerOnHold {
+					inputStatuses[i] = HeldStatus
+				}
+				continue
+			}
+			inputStatuses[i] = PressedStatus
+		} else {
+			inputStatuses[i] = OffStatus
+		}
+	}
+
+	return inputStatuses, nil
+}
+
+// LinkLeds link the behavior of the LEDs to the touch sensors.
+// Doing so, disabled the option for the host to set specific LEDs on/off.
+func (d *Dev) LinkLeds() error {
+	if err := d.regWrapper.WriteUint8(reg_LEDLinking, 0xff); err != nil {
+		return wrap(fmt.Errorf("failed to link LEDs - %s", err))
+	}
+	d.opts.LinkedLEDs = true
+	return nil
+}
+
+// UnlinkLeds disassociate the LEDs from the input sensors allowing the host to
+// control the LEDs.
+func (d *Dev) UnlinkLeds() error {
+	if err := d.regWrapper.WriteUint8(reg_LEDLinking, 0x00); err != nil {
+		return wrap(fmt.Errorf("failed to unlink LEDs - %s", err))
+	}
+	d.opts.LinkedLEDs = false
+	return nil
+}
+
+// AllLedsOn turns all the LEDs on.
+// This is quite more efficient than looping through each led and turn them on.
+func (d *Dev) AllLedsOn() error {
+	if d.opts.LinkedLEDs {
+		return wrap(fmt.Errorf("can't manually set LEDs when they are linked to sensors"))
+	}
+	return d.regWrapper.WriteUint8(reg_LEDOutputControl, 0xff)
+}
+
+// AllLedsOff turns all the LEDs off.
+// This is quite more efficient than looping through each led and turn them off.
```

Comment:
Always put an empty line after the first sentence, this changes godoc output format.